.circleci/config.yml

```diff
@@ -234,13 +262,31 @@ jobs:
       # Not using a workspace here as Node and Yarn versions
       # differ between the macOS image and the Docker containers above.
       - run:
+          name: Pre-start simulator
+          command: xcrun simctl boot "iPhone 7" || true
+      - <<: *restore-brew-cache
+      - run:
+          name: Install HomeBrew dependencies
+          command: |
+              brew tap homebrew/bundle
+              brew bundle
+      - <<: *save-brew-cache
+      - run:
+          name: Set npm version and install global npm dependencies
           command: |
-              brew update
-              brew tap wix/brew
-              brew install --HEAD applesimutils
-              yarn global add detox-cli
-              yarn global add react-native-cli
-      - run: yarn
+              npm i -g n
+              sudo n 9.3
+              npm i -g detox-cli
+              npm i -g react-native-cli
+      - restore_cache:
+          key: osx-dependencies-{{ checksum "package.json" }}
+      - run:
+          name: Install dependencies
+          command: npm install
+      - save_cache:
+          paths:
+            - /Users/distiller/project/node_modules
+          key: osx-dependencies-{{ checksum "package.json" }}
```

Comment:
It looks like you use the same cache for `get-cache-osx` and `save-cache-osx` steps (except the path to package.json). Is this intended?

Reply:
The working directory is different for e2e as it’s run in the project root. I couldn’t see a way to pass the key and they need to coincide otherwise it would be a different cache for e2e and build jobs. Which isn’t too bad but because only e2e uses this key then there wasn’t much point in pulling out to an alias so I’ve left it inside the job